packages/embedder/workflows/embedMetadata.ts

```diff
@@ -61,39 +63,40 @@ export const embedMetadata: JobQueueStepRun<
       forceBuildText: true
     })
   }
+  // Delete existing chunks for this metadataID to prevent stale data
+  await pg
+    // cast to any because these types won't be available in CI
+    .deleteFrom(embeddingModel.tableName as any)
+    .where('embeddingsMetadataId', '=', embeddingsMetadataId)
+    .execute()
+
   // Cannot use summarization strategy if generation model has same context length as embedding model
   // We must split the text & not tokens because BERT tokenizer is not trained for linebreaks e.g. \n\n
-  const errors = await Promise.all(
-    chunks.map(async (chunk, chunkNumber) => {
-      const embeddingVector = await embeddingModel.getEmbedding(chunk)
-      if (embeddingVector instanceof Error) {
-        return new JobQueueError(
-          `unable to get embeddings: ${embeddingVector.message}`,
-          ms('1m'),
-          10,
-          {forceBuildText: true}
-        )
-      }
-      await pg
-        // cast to any because these types won't be available in CI
-        .insertInto(embeddingModel.tableName)
-        .values({
-          // TODO is the extra space of a null embedText really worth it?!
-          embedText: chunks.length > 1 ? chunk : null,
-          embedding: numberVectorToString(embeddingVector),
-          embeddingsMetadataId,
-          chunkNumber: chunks.length > 1 ? chunkNumber : null
+  const errors: (JobQueueError | undefined)[] = []
+
+  for (let i = 0; i < chunks.length; i++) {
+    const chunk = chunks[i]!
+    const embeddingVector = await embeddingModel.getEmbedding(chunk, {priority: 'low'})
+    if (embeddingVector instanceof Error) {
+      errors.push(
+        new JobQueueError(`unable to get embeddings: ${embeddingVector.message}`, ms('1m'), 10, {
+          forceBuildText: true
         })
-        .onConflict((oc) =>
-          oc.columns(['embeddingsMetadataId', 'chunkNumber']).doUpdateSet((eb) => ({
-            embedText: eb.ref('excluded.embedText'),
-            embedding: eb.ref('excluded.embedding')
-          }))
-        )
-        .execute()
-      return undefined
-    })
-  )
+      )
+      continue
+    }
+    await pg
+      // cast to any because these types won't be available in CI
+      .insertInto(embeddingModel.tableName as any)
+      .values({
+        embedText: chunk,
+        tsv: sql`to_tsvector('english', ${chunk})`, // TODO: other languages
```

Comment:
- 1 `inferLanguage` has a lookup table for ISO 639-3 to ISO 639-1.

Create a lookup table for ISO639-1 to cfgname. See https://gist.github.com/Josantonius/b455e315bc7f790d14b136d61d9ae469#file-language-codes-json

on startup, run `SELECT cfgname FROM pg_ts_config;`
then, filter your lookup table values with the the result of the query. This is because we don't know if PPMIs who control their own PG will have the same language files as we have on our hosted google instance.

then, when we infer language, we can bail early if the language is unsupported (or default to english)